### PR TITLE
Optimize some SSE2 intrinsic importation and re-enable tests

### PR DESCRIPTION
**Pull request by fiigii**
This PR 
1. re-enables the hardware intrinsic tests that were disabled in https://github.com/dotnet/coreclr/pull/19949 and moves to template-based.
2. optimize some SSE2 intrinsic importation by removing unnecessary `HW_Flag_BaseTypeFromFirstArg`, which avoid calling into the expensive VM functions.

### CODE REVIEW

**Comment by fiigii** [comment]
@CarolEidt @tannergooding PTAL

File: tests/src/JIT/HardwareIntrinsics/X86/Popcnt/Popcnt.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using System;
```
Comment (tannergooding) [inline tests/src/JIT/HardwareIntrinsics/X86/Popcnt/Popcnt.cs]: It would be nice if we could just move this to be templated.

Comment (fiigii) [reply tests/src/JIT/HardwareIntrinsics/X86/Popcnt/Popcnt.cs]: This test will be changed again after exposing 64-bit intrinsic in nested classes. Let’s move them later.

**Comment by fiigii** [comment]
Can we merge this?

**Comment by tannergooding** [comment]
It would be nice if @CarolEidt could give this a cursory glance as well.

**Comment by fiigii** [comment]
Thanks, I will move all the scalar intrinsic tests to the template after https://github.com/dotnet/coreclr/pull/20146 gets done.

**Comment by fiigii** [comment]
@CarolEidt ping?

**Pull request merged**